main.rb

```diff
@@ -21,55 +23,47 @@
     exit 0
 end
 
+
 # if File.exist?(cocoapods_project_path)
 #     puts "Pods already installed."
 #     exit 0
+# else
+#     puts "Pods does not exist. Pod install command is being executed"
 # end
 
+
 def runCommand(command)
     puts "@@[command] #{command}"
     unless system(command)
       exit $?.exitstatus
     end
 end
 
-if cocoapods_version.nil?
-  if File.exist?(cocoapods_podfilelock_path)
-    versionArray = File.read(cocoapods_podfilelock_path).scan(/(?<=COCOAPODS: )(.*)/)[0]
-    if versionArray && versionArray[0]
-      cocoapods_version = versionArray[0]
-      puts "Podfile.lock version = #{cocoapods_version}"
-      if `which rbenv`.empty?
-        runCommand("sudo gem install cocoapods -v #{cocoapods_version} --no-document")
-      else
-        runCommand("gem install cocoapods -v #{cocoapods_version} --no-document")
-      end
-    end
-  end
-else
-  puts "Cocoapods version = #{cocoapods_version}"
-  if `which rbenv`.empty?
-    runCommand("sudo gem install cocoapods -v #{cocoapods_version} --no-document")
-  else
-    runCommand("gem install cocoapods -v #{cocoapods_version} --no-document")
+def podInstall(cocoapods_version, project_dir_path)
+  cocoapods_cmd_prefix = cocoapods_version ? "pod _#{cocoapods_version}_" : "pod"
+
+  Dir.chdir(project_dir_path) do
+    runCommand("#{cocoapods_cmd_prefix} install")
   end
 end
 
-unless cocoapods_version.nil?
-  runCommand("pod _#{cocoapods_version}_ setup")
-  runCommand("pod _#{cocoapods_version}_ repo update")
-else
-  runCommand("pod --version")
+if cocoapods_version.nil?
+  puts "Using System Default pod version."
+  runCommand('pod --version')
   runCommand("pod repo update")
+  Dir.chdir(project_dir_path) do
+    runCommand("pod install")
+  end
+else
+    puts "Cocoapods version = #{cocoapods_version}"
+        if `which rbenv`.empty?
```

Comment:
@boztopuz For all our runner pools, `which rbenv.empty?` will return `false` since they have `rbenv` installed. IMO, the condition was for backward compatibility for that time when `rbenv` had been developed.

I'm handling the case. :raising_hand_man: